### PR TITLE
Refactor monitoring

### PR DESCRIPTION
**Pull request by qemanuel**
### Added
- aws-iam-authenticator is not needed anymore
- loki-stack has been replaced for loki-distributed, kube-stack-prometheus and fluent-bit helm charts
- tempo-distributed helm chart
- cert-manager helm chart
- opentelemetry manifests for auto-instrumentation
- ingress-nginx, metrics-server and cluster-autoscaler helm charts updated to latest version

### CODE REVIEW

File: helm.tf [hunk 1]
```diff
@@ -34,21 +8,30 @@ resource "helm_release" "ingress_nginx" {
   create_namespace  = true
   repository        = "https://kubernetes.github.io/ingress-nginx"
   chart             = "ingress-nginx"
-  version           = "3.34.0"
+  version           = "4.0.18"
```
Comment (ivours) [inline helm.tf:11]: We should allow chart version selection

File: helm.tf [hunk 2]
```diff
@@ -60,7 +43,7 @@ resource "helm_release" "cluster_autoscaler" {
   namespace  = "kube-system"
   repository = "https://kubernetes.github.io/autoscaler"
   chart      = "cluster-autoscaler"
-  version    = "9.9.2"
+  version    = "9.16.1"
```
Comment (ivours) [inline helm.tf:46]: We should allow chart version selection (module input)

File: helm.tf [hunk 3]
```diff
@@ -83,7 +66,7 @@ resource "helm_release" "metrics_server" {
   namespace  = "kube-system"
   repository = "https://charts.bitnami.com/bitnami"
   chart      = "metrics-server"
-  version    = "5.8.11"
+  version    = "5.11.3"
```
Comment (ivours) [inline helm.tf:69]: We should allow chart version selection (module input)

File: helm.tf [hunk 4]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
 
 }
 
+resource "helm_release" "cert_manager" {
+  count      = var.helm_cert_manager_enabled || var.k8s_opentelemetry_enabled ? 1 : 0
+  name       = "cert-manager"
+  namespace  = "cert-manager"
+  repository = "https://charts.jetstack.io"
+  chart      = "cert-manager"
+  version    = "1.6.1"
```
Comment (ivours) [inline helm.tf:91]: We should allow chart version selection (module input)

File: helm.tf [hunk 5]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+# ========================= monitoring charts ========================= #
+
+resource "helm_release" "prometheus_stack" {
+  count             = var.helm_prometheus_enabled ? 1 : 0
+  name              = "prometheus-stack"
+  namespace         = "monitoring"
+  create_namespace  = true
+  repository        = "https://prometheus-community.github.io/helm-charts"
+  chart             = "kube-prometheus-stack"
+  version           = "35.0.3"
```
Comment (ivours) [inline helm.tf:111]: We should allow chart version selection (module input)

File: helm.tf [hunk 6]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+resource "helm_release" "loki_distributed" {
+  count             = var.helm_loki_enabled ? 1 : 0
+  name              = "loki-distributed"
   namespace         = "monitoring"
   create_namespace  = true
   repository        = "https://grafana.github.io/helm-charts"
-  chart             = "loki-stack"
-  version           = "2.4.1"
+  chart             = "loki-distributed"
+  version           = "0.48.3"
```
Comment (ivours) [inline helm.tf:218]: We should allow chart version selection (module input)

File: helm.tf [hunk 7]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
-      name  = "grafana.grafana.ini.server.domain"
-      value = var.grafana_ingress_host
+      name  = "queryFrontend.resources.limits.cpu"
+      value = var.loki_query_frontend_limits_cpu
     }
   }
 
   dynamic "set" {
-    for_each = var.grafana_ingress_enabled ? ["do it"] : []
+    for_each = var.loki_query_frontend_limits_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:437]: I would name this variable `loki_query_frontend_limits_memory`

File: helm.tf [hunk 8]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
     content {
-      name  = "grafana.grafana.ini.server.root_url"
-      value = "https://${var.grafana_ingress_host}"
+      name  = "queryFrontend.resources.limits.memory"
+      value = var.loki_query_frontend_limits_ram
```
Comment (ivours) [inline helm.tf:440]: I would name this variable `loki_query_frontend_limits_memory`

File: helm.tf [hunk 9]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+      name  = "compactor.resources.requests.cpu"
+      value = var.loki_compactor_requests_cpu
+    }
   }
 
+  dynamic "set" {
+    for_each = var.loki_compactor_requests_ram != null ? ["do it"] : []
+    content {
+      name  = "compactor.resources.requests.memory"
+      value = var.loki_compactor_requests_ram
```
Comment (ivours) [inline helm.tf:463]: I would name this variable `loki_compactor_requests_memory`

File: helm.tf [hunk 10]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+  dynamic "set" {
+    for_each = var.loki_compactor_requests_cpu != null ? ["do it"] : []
+    content {
+      name  = "compactor.resources.requests.cpu"
+      value = var.loki_compactor_requests_cpu
+    }
   }
 
+  dynamic "set" {
+    for_each = var.loki_compactor_requests_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:460]: I would name this variable loki_compactor_requests_memory

File: helm.tf [hunk 11]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+      name  = "queryFrontend.resources.requests.cpu"
+      value = var.loki_query_frontend_requests_cpu
+    }
   }
 
+  dynamic "set" {
+    for_each = var.loki_query_frontend_requests_ram != null ? ["do it"] : []
+    content {
+      name  = "queryFrontend.resources.requests.memory"
+      value = var.loki_query_frontend_requests_ram
```
Comment (ivours) [inline helm.tf:424]: I would name this variable `loki_query_frontend_requests_memory`

File: helm.tf [hunk 12]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+  dynamic "set" {
+    for_each = var.loki_query_frontend_requests_cpu != null ? ["do it"] : []
+    content {
+      name  = "queryFrontend.resources.requests.cpu"
+      value = var.loki_query_frontend_requests_cpu
+    }
   }
 
+  dynamic "set" {
+    for_each = var.loki_query_frontend_requests_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:421]: I would name this variable `loki_query_frontend_requests_memory`

File: helm.tf [hunk 13]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+  dynamic "set" {
+    for_each = var.loki_compactor_limits_cpu != null ? ["do it"] : []
+    content {
+      name  = "compactor.resources.limits.cpu"
+      value = var.loki_compactor_limits_cpu
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.loki_compactor_limits_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:476]: I would name this variable `loki_compactor_limits_memory`

File: helm.tf [hunk 14]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    content {
+      name  = "compactor.resources.limits.memory"
+      value = var.loki_compactor_limits_ram
```
Comment (ivours) [inline helm.tf:479]: I would name this variable `loki_compactor_limits_memory`

File: helm.tf [hunk 15]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    for_each = var.loki_index_gateway_requests_cpu != null ? ["do it"] : []
+    content {
+      name  = "indexGateway.resources.requests.cpu"
+      value = var.loki_index_gateway_requests_cpu
+    }
   }
 
-# grafana - sidecard
+  dynamic "set" {
+    for_each = var.loki_index_gateway_requests_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:520]: I would name this variable `loki_index_gateway_requests_memory`

File: helm.tf [hunk 16]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    content {
+      name  = "indexGateway.resources.requests.memory"
+      value = var.loki_index_gateway_requests_ram
```
Comment (ivours) [inline helm.tf:523]: I would name this variable `loki_index_gateway_requests_memory`

File: helm.tf [hunk 17]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+  dynamic "set" {
+    for_each = var.loki_index_gateway_limits_cpu != null ? ["do it"] : []
+    content {
+      name  = "indexGateway.resources.limits.cpu"
+      value = var.loki_index_gateway_limits_cpu
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.loki_index_gateway_limits_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:536]: I would name this variable `loki_index_gateway_limits_memory`

File: helm.tf [hunk 18]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    content {
+      name  = "indexGateway.resources.limits.memory"
+      value = var.loki_index_gateway_limits_ram
```
Comment (ivours) [inline helm.tf:539]: I would name this variable `loki_index_gateway_limits_memory`

File: helm.tf [hunk 19]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    for_each = var.loki_gateway_requests_cpu != null ? ["do it"] : []
+    content {
+      name  = "gateway.resources.requests.cpu"
+      value = var.loki_gateway_requests_cpu
+    }
   }
 
-# ========================= prometheus ========================= #
+  dynamic "set" {
+    for_each = var.loki_gateway_requests_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:574]: I would name this variable `loki_gateway_requests_memory`

File: helm.tf [hunk 20]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    content {
+      name  = "gateway.resources.requests.memory"
+      value = var.loki_gateway_requests_ram
```
Comment (ivours) [inline helm.tf:577]: I would name this variable `loki_gateway_requests_memory`

File: helm.tf [hunk 21]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+  dynamic "set" {
+    for_each = var.loki_gateway_limits_cpu != null ? ["do it"] : []
+    content {
+      name  = "gateway.resources.limits.cpu"
+      value = var.loki_gateway_limits_cpu
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.loki_gateway_limits_ram != null ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:590]: I would name this variable `loki_gateway_limits_memory`

File: helm.tf [hunk 22]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+    content {
+      name  = "gateway.resources.limits.memory"
+      value = var.loki_gateway_limits_ram
```
Comment (ivours) [inline helm.tf:593]: I would name this variable `loki_gateway_limits_memory`

File: helm.tf [hunk 23]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+# ================== fluent-bit ================== #
+
+resource "helm_release" "fluent_bit" {
+  count             = var.helm_fluent_bit_enabled ? 1 : 0
+  name              = "fluent-bit"
+  namespace         = "monitoring"
+  create_namespace  = true
+  repository        = "https://fluent.github.io/helm-charts"
+  chart             = "fluent-bit"
+  version           = "0.19.24"
```
Comment (ivours) [inline helm.tf:634]: Allow version selection (module input)

File: helm.tf [hunk 24]
```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
...
+# ================== tempo-distributed ================== #
+
+resource "helm_release" "tempo_distributed" {
+  count             = var.helm_tempo_enabled ? 1 : 0
+  name              = "tempo-distributed"
+  namespace         = "monitoring"
+  create_namespace  = true
+  repository        = "https://grafana.github.io/helm-charts"
+  chart             = "tempo-distributed"
+  version           = "0.17.1"
```
Comment (ivours) [inline helm.tf:653]: Allow version selection (module input)

**Review by ivours — CHANGES_REQUESTED** [review]
Added comments regarding chart versions.

Please, also replace the **ram** word with **memory**.

File: helm.tf (hunk 1 above)
Comment (qemanuel) [reply helm.tf:11]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 2 above)
Comment (qemanuel) [reply helm.tf:46]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 3 above)
Comment (qemanuel) [reply helm.tf:69]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 4 above)
Comment (qemanuel) [reply helm.tf:91]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 5 above)
Comment (qemanuel) [reply helm.tf:111]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 6 above)
Comment (qemanuel) [reply helm.tf:218]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 23 above)
Comment (qemanuel) [reply helm.tf:634]: I'm marking this as resolved because we agreed to don't do it

File: helm.tf (hunk 24 above)
Comment (qemanuel) [reply helm.tf:653]: I'm marking this as resolved because we agreed to don't do it

**Comment by qemanuel** [comment]
> Added comments regarding chart versions.
> 
> Please, also replace the **ram** word with **memory**.

-  Replacements done
- Charts versions will be treated in a future

**Review by ivours — APPROVED** [review]

**Pull request merged**